### PR TITLE
MyOffers active contracts

### PR DESCRIPTION
**Pull request by farinalvaro**
## Features
- adds ability to fetch agreements by consumer or provider
- adds set filters action in agreements reducer
- adds agreements views for providers and customers

### CODE REVIEW

**Review by julianmrodri — APPROVED** [review]
LGTM

**Review by julianmrodri — APPROVED** [review]
LGTM

**Review by jurajpiar — CHANGES_REQUESTED** [review]
Nice job! Just remove the unnecessary check in agreements _fetch, and in the Context make the filters object optional, which should replace the account check in the fetching useEffect.

File: src/components/organisms/storage/myoffers/ExpandableOffer.tsx [hunk 1]
```diff
@@ -55,19 +60,39 @@ const ExpandableOffer: FC<ExpandableOfferProps> = ({
...
 }) => {
   const classes = useStyles()
+  const {
+    state: { account },
+  } = useContext(Web3Store)
+  const {
+    state: {
+      agreements,
+    },
+    dispatch: agreementsDispatch,
```
[the remark this answers is not in the document]
Comment (farinalvaro) [reply src/components/organisms/storage/myoffers/ExpandableOffer.tsx:70]: well, to be honest I just found it easier to read when using it 😅

File: src/context/Services/storage/agreements/Context.tsx [hunk 2]
```diff
@@ -12,6 +12,10 @@ import actions from './actions'
 export const initialState: State = {
   contextID: 'storage_agreements',
   agreements: [],
+  filters: {
+    consumer: '',
+    provider: undefined,
```
[the remark this answers is not in the document]
Comment (farinalvaro) [reply src/context/Services/storage/agreements/Context.tsx]: fixed, thanks

File: src/context/Services/storage/agreements/Context.tsx [hunk 3]
```diff
@@ -84,7 +90,7 @@ export const Provider: FC = ({ children }) => {
           })
         })
     }
-  }, [isInitialised, api, account, appDispatch])
+  }, [isInitialised, api, account, appDispatch, filters])
```
[the remark this answers is not in the document]
Comment (farinalvaro) [reply src/context/Services/storage/agreements/Context.tsx]: fixed, thanks

**Review by jurajpiar — APPROVED** [review]
Sweet as

**Pull request merged**